### DISCUSSION TITLE
Deleted Dremio JDBC connection - need to restore

### DISCUSSION THREAD

**Discussion by seg729** [dbeaver/dbeaver#20626] [Q&A] [answered]
In the "Connect to a database" dialogue, I managed to delete an existing Dremio connection I had.  This deleted the icon and the default JDBC driver.  I'm not able to restore the connection or the driver.  Is there a way to restore or reset back to the original installation drivers and icons or do I need to uninstall and re-install DBeaver?  Any help would be appreciated.  The screenshot below shows the original setup which I managed to delete and now I'm trying to restore.

Setup:
MacOS Ventura 13.4
Dbeaver Community Edition Version 23.1.2

<img width="686" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/47a0cbdd-17cc-4645-b41b-f5fdffafaf8b">

Driver information
<img width="276" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/99d1fa5a-0578-475b-8dc6-52a824d72190">

**Comment by E1izabeth** [comment]
Do you have Dremio in Driver Manager? You can find Driver Manager in the main menu Database->Driver Manager.
![image](https://github.com/dbeaver/dbeaver/assets/28875055/1c367b8d-69df-4a40-98a7-8a0f4687c03c)

Could you explain what do you mean by "delete an existing Dremio connection"? Does it disapeared from you navigator or you can't find it Connect to a database dialog? AFAIK there is no way to delete database driver configuration from the Connect to a database dialog.

**Comment by seg729** [comment]
Thanks for the reply.

From Driver Manager, in my installation, the Dremio driver was showing ‘Unavailable’. I clicked on ‘Un-delete’ but nothing happened.  I then clicked on ‘Delete’ (believing I could click on ‘Un-delete’ again, and that got rid of the icon, configuration, and the JDBC driver.  I tried to re-install using the ‘Connect to a Database’ dialog but now the Dremio icon was missing.  That’s when I realized my ‘Delete’ action also deleted the JDBC driver from my local machine.

I was able to create a user-defined connection in Driver Manager by copying the Dremio JDBC folder from a colleague’s Windows computer and pasting it into mine (same path as before).  That provided me a way to connect but did not restore the Dremio icon (dolphin) on the connection.  Since I don’t know what else I might have messed up, I want to see if I can restore the drivers/settings to the out-of-the-box connection state.

Is there a way to restore all this to the original settings/drivers/icons?  I got this working again, but I’m worried of what else I messed up.

Thanks!

<img width="439" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/d44f0819-5db0-41dd-83b6-f3762179505c">
<img width="586" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/06351e3d-455e-4c01-9ed9-18646f763bd4">
<img width="1268" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/0d732623-1949-4d91-a77c-0388830ef9d6">
<img width="500" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/ff2e91a3-ef33-440a-befd-28ac3593c060">
<img width="570" alt="image" src="https://github.com/dbeaver/dbeaver/assets/140108645/31cf8442-13bf-4076-a574-6ed0340a0286">



From: Elizabeth ***@***.***>
Reply-To: dbeaver/dbeaver ***@***.***>
Date: Thursday, July 20, 2023 at 3:12 PM
To: dbeaver/dbeaver ***@***.***>
Cc: Saul Garcia-External ***@***.***>, Author ***@***.***>
Subject: [External] Re: [dbeaver/dbeaver] Deleted Dremio JDBC connection - need to restore (Discussion #20626)


EXTERNAL MESSAGE
________________________________


Do you have Dremio in Driver Manager? You can find Driver Manager in the main menu Database->Driver Manager.
[image]<https://urldefense.com/v3/__https:/user-images.githubusercontent.com/28875055/255019749-1c367b8d-69df-4a40-98a7-8a0f4687c03c.png__;!!ODpDvJZr5w!GWk8SEIwHgj11znSKk32GVGlt_Vv5USSTS6u4CQPH9B0jLxEWHHeCP_pThLWchmX8JGml2ZVKS-3LB9ThXeDYx9uXpY$>

Could you explain what do you mean by "delete an existing Dremio connection"? Does it disapeared from you navigator or you can't find it Connect to a database dialog? AFAIK there is no way to delete database driver configuration from the Connect to a database dialog.

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/orgs/dbeaver/discussions/20626*discussioncomment-6503875__;Iw!!ODpDvJZr5w!GWk8SEIwHgj11znSKk32GVGlt_Vv5USSTS6u4CQPH9B0jLxEWHHeCP_pThLWchmX8JGml2ZVKS-3LB9ThXeDqPe-LOg$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BBM6GZLMFSCIICOGG4ZZ4NTXRGUMVANCNFSM6AAAAAA2R34IHY__;!!ODpDvJZr5w!GWk8SEIwHgj11znSKk32GVGlt_Vv5USSTS6u4CQPH9B0jLxEWHHeCP_pThLWchmX8JGml2ZVKS-3LB9ThXeDHeUJM3o$>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

******************************************************************** 
This e-mail and any attachment hereto, is intended only for use by the addressee(s) named above and may contain legally privileged and/or confidential information. If you are not the intended recipient of this e-mail, any dissemination, distribution or copying of this email, or any attachment hereto, is strictly prohibited. If you receive this email in error please immediately notify me by return electronic mail and permanently delete this email and any attachment hereto, any copy of this e-mail and of any such attachment, and any printout thereof. Finally, please note that only authorized representatives of Regeneron Pharmaceuticals, Inc. have the power and authority to enter into business dealings with any third party. 
********************************************************************

**Reply by E1izabeth** [reply, marked as answer]
Don't you see Dremio in the list to restore when pressing Un-delete button?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/96cbf670-3483-4085-910f-426590332b51)

**Reply by seg729** [reply]
Yes!  Using the 'Un-delete' from Driver Manager worked 100% and restored everything!  I forced download on the driver and was able to connect via the 'Connect to a database' wizard.  The icon was restored, and the connection was successful.

I was convinced I had unsuccessfully tried the 'Un-delete' button when I had done the 'Delete' before, but I guess I just imagined it.  

Thank you for your help!  This can be marked as 'Answered'

**Comment by E1izabeth** [comment]
Resetting workspace configuration might help, but your existing connections and settings may disappear. You can find it in Help -> Reset Settings.. But I'd try to figure out what's going on first, maybe I could help you to fix it if I realize what exactly went wrong with the driver in you workspace.
![image](https://github.com/dbeaver/dbeaver/assets/28875055/8cf74d8d-745c-43d0-9d7a-8c8bbbe0a336)
